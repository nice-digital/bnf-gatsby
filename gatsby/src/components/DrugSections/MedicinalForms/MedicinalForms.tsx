import { Link } from "gatsby";
import { type FC } from "react";

import { type MedicinalFormsWithSlugs, type PotWithSlug } from "src/types";

import styles from "../DrugSection.module.scss";

export type MedicinalFormsProps = PotWithSlug &
	MedicinalFormsWithSlugs & { drugSlug: string; drugTitle: string };

export const MedicinalForms: FC<MedicinalFormsProps> = ({
	potName,
	slug,
	medicinalForms,
	initialStatement,
	specialOrderManufacturersStatement,
	drugSlug,
	drugTitle,
}) => {
	return (
		<section aria-labelledby={slug} className={styles.section}>
			<h2 id={slug} dangerouslySetInnerHTML={{ __html: potName }} />
			<p dangerouslySetInnerHTML={{ __html: initialStatement }} />
			{specialOrderManufacturersStatement && (
				<p
					dangerouslySetInnerHTML={{
						__html: specialOrderManufacturersStatement,
					}}
				/>
			)}
			{medicinalForms.length > 0 ? (
				<>
					<p>
						View{" "}
						<Link
							to={`/drugs/${drugSlug}/medicinal-forms/`}
							id="medicinal-forms-link"
						>
							<span dangerouslySetInnerHTML={{ __html: drugTitle }} /> medicinal
							forms and pricing information
						</Link>{" "}
						or jump straight to:
					</p>
					<ul aria-labelledby="medicinal-forms-link">
						{medicinalForms.map(({ form, slug }) => (
							<li key={form}>
								<Link to={`/drugs/${drugSlug}/medicinal-forms/#${slug}`}>
									{form}
								</Link>
							</li>
						))}
					</ul>
				</>
			) : null}
		</section>
	);
};
