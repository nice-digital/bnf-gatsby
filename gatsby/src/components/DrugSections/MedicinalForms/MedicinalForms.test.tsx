import { render, screen } from "@testing-library/react";

import { MedicinalForms, type MedicinalFormsProps } from "./MedicinalForms";

describe("MedicinalForms", () => {
	const props: MedicinalFormsProps = {
		drug: {
			slug: "anti-d-rh0-immunoglobulin",
			title: "anti-d (rh<sub>0</sub>) immunoglobulin",
		},
		potName: "Medicinal forms",
		slug: "medicinal-forms",
		initialStatement: "<em>No</em> licensed medicines listed.",
		specialOrderManufacturersStatement: null,
		medicinalForms: [
			{
				form: "Tablets",
				slug: "tablets",
			},
			{
				form: "Eye drops",
				slug: "eye-drops",
			},
		],
	};

	it("should render section with accessible name", () => {
		render(<MedicinalForms {...props} />);

		expect(
			screen.getByRole("region", { name: "Medicinal forms" })
		).toBeInTheDocument();
	});

	it("should render heading 2 with section name", () => {
		render(<MedicinalForms {...props} />);

		expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
			"Medicinal forms"
		);
	});

	it("should match snapshot", () => {
		const { container } = render(<MedicinalForms {...props} />);

		expect(container).toMatchSnapshot();
	});
});
