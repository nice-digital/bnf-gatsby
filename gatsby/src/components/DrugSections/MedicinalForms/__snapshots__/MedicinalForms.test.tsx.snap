// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicinalForms should match snapshot 1`] = `
<div>
  <section
    aria-labelledby="medicinal-forms"
    class="section"
  >
    <h2
      id="medicinal-forms"
    >
      Medicinal forms
    </h2>
    <p>
      <em>
        No
      </em>
       licensed medicines listed.
    </p>
    <p>
      <a
        href="/drugs/anti-d-rh0-immunoglobulin/medicinal-forms/"
      >
        View medicinal forms and pricingÂ information
      </a>
    </p>
    <p>
      Or jump straight to:
    </p>
    <ul
      aria-label="Links to each medicinal form"
    >
      <li>
        <a
          href="/drugs/anti-d-rh0-immunoglobulin/medicinal-forms/#tablets"
        >
          Tablets
        </a>
      </li>
      <li>
        <a
          href="/drugs/anti-d-rh0-immunoglobulin/medicinal-forms/#eye-drops"
        >
          Eye drops
        </a>
      </li>
    </ul>
  </section>
</div>
`;
